hs/src/Reach/Verify/SMT.hs

```diff
@@ -140,6 +140,7 @@ instance Monoid a => Monoid (App a) where
 data SMTMapInfo = SMTMapInfo
```

Comment:
SMT is going to confuse arrays and maps keyed on numbers. Write some tests that generate errors and you'll see this and figure out a solution

Reply:
I think this actually won't happen because when we generate the SMT arrays, we use SMT's `Int` type for `Array`s & and theres no Reach type that converts to an SMT `Int`, so Map would never use it

Reply:
`UInt` (and `UInt256`) are `Int`. They are just type aliases (called sort aliases) for `Int` and when we look at the model, they are expressed as `Int`, not `UInt`